lib/modules/dosomething/dosomething_reportback/includes/ReportbackItemTransformer.php

```diff
@@ -40,11 +40,21 @@ public function index($parameters) {
   public function show($id) {
     try {
       $reportbackItem = ReportbackItem::get($id);
-      $reportbackItem = $this->removeUnauthorizedResults($reportbackItem);
+
+      if (!user_access('administer modules')) {
+        $reportbackItem = $this->removeUnauthorizedResults($reportbackItem, TRUE);
+      }
+
       $reportbackItem = services_resource_build_index_list($reportbackItem, 'reportback-items', 'id');
     }
     catch (Exception $error) {
-      http_response_code('404');
+      if ($error->getMessage() === 'Access denied.') {
+        http_response_code('403');
+      }
+      else {
+        http_response_code('404');
+      }
```

Comment:
It was causing an issue because when hitting the index of `/signups`, a `404` status code was popping up in addition to signup objects. This was because it was looking for kudos associated with reportbacks that were being returned in the signup object - if there were no kudos, a `404` status code was returned in addition to signups so it made this very confusing. Chatting more with @angaither and @DFurnes , we decided to avoid this, the error codes should be returned at the transformer level. LMK if this is still confusing bc I'm not the best at explaining!